kpops/components/base_components/kafka_connect.py

```diff
@@ -34,6 +34,9 @@
 
 class KafkaConnector(PipelineComponent, ABC):
     type: str = "kafka-connect"
+    schema_type: Literal["kafka-connect"] = Field(  # type: ignore[assignment]
```

Comment:
I noticed that the class name differs from the `type` and `schema_type` names. I would rather have it `KafkaConnect`. @disrupted WDYT?

Reply:
I think it makes more sense to rename the type to `kafka-connector`, but that would be a breaking change. We could deprecate it and use it as an alias for some time (normal is 12 months I think)?

`kafka-connector` being the parent of all specific connectors named `kafka-{type}-connector` is a lot more intuitive imo

Reply:
yeah, these should be uniform. However, I would propose to change `type` (and `schema_type`) to `kafka-connector`.

The class name reads nicer as `KafkaConnector` imo with the subclasses `KafkaSinkConnector` and `KafkaSourceConnector`

Reply:
that would be a breaking change though

Reply:
@sujuka99 good point. lucky for us we're pre-1.0 so we don't have to stick to deprecation plans 😄 
@raminqaf good plan!

Reply:
#149